src/scripting/game_lua_kernel.cpp

```diff
@@ -3064,6 +3065,41 @@ int game_lua_kernel::intf_play_sound(lua_State *L)
 	return 0;
 }
 
+/**
+ * Sets an achievement as being completed.
+ * - Arg 1: string - content_for.
+ * - Arg 2: string - id.
+ */
+int game_lua_kernel::intf_set_achievement(lua_State *L)
+{
+	const char *content_for = luaL_checkstring(L, 1);
+	const char *id = luaL_checkstring(L, 2);
+	preferences::set_achievement(content_for, id);
+
+	return 0;
+}
+
+/**
+ * Returns whether an achievement has been completed.
+ * - Arg 1: string - content_for.
+ * - Arg 2: string - id.
+ * - Ret 1: boolean.
+ */
+int game_lua_kernel::intf_has_achievement(lua_State *L)
+{
+	const char *content_for = luaL_checkstring(L, 1);
+	const char *id = luaL_checkstring(L, 2);
+
+	if(resources::controller->is_networked_mp() && synced_context::is_synced()) {
```

Comment:
Commenting to test later - is `is_synced` the right condition? I'd assume we'd want this fallback anytime that we're not in a synchronised choice, which IIUC is different to being in a synchronised event.

Reply:
If not, then https://github.com/wesnoth/wesnoth/blob/0eb6ecb42e3c9a3c1a283745441cf58c4390a255/src/side_filter.cpp#L220 also needs to be fixed.